core/src/robot/Robot.cc

```diff
@@ -1188,6 +1253,30 @@ namespace jiminy
         }
     }
 
+    void Robot::computeConstraints(vectorN_t const & q,
+                                   vectorN_t const & v,
+                                   matrixN_t & jacobianOut,
+                                   vectorN_t & driftOut)
+    {
+        // Give the right number of columns to the jacobian.
+        jacobianOut.resize(0, pncModel_.nv);
+
+        // Call computeAllTerms to update frame jacobian: this is needed by forwardDynamics anyway.
+        pinocchio::computeAllTerms(pncModel_, pncData_, q, v);
+
+        for (auto & constraint : constraintsHolder_)
+        {
+            matrixN_t J = constraint.second->getJacobian(q);
+            vectorN_t drift = constraint.second->getDrift(q, v);
+            int constraintSize = drift.size();
+            // Concatenate drift and jacobians.
+            jacobianOut.conservativeResize(jacobianOut.rows() + constraintSize, Eigen::NoChange);
```

Comment:
This is a point of debate: do we allow the user to change the constraints as the simulation go ? 

Drawback: you don't know in advance the number of constraints, so you have to do dynamic resizing of the matrices, which probably slows down a bit the simulation.

Advantage: even if it's not supported right now, keeping it as is would enable the use of analytical contacts for intermittent constraints: for instance, a walking robot. We "just" need a collision function to determine whether or not the constraint is active at a given time.

Reply:
I don't know. Perhaps having a jacobian buffer and checking if the size has changed before doing the conservative resize is the solution. This way, the costly memory allocation will be performed only if necessary, thereby combining the best of both solutions.

Reply:
The thing is I don't know the size I will have as input.

In the context of ConstraintFunctor, I don't want the user to have to define a third method (along get_jacobian and get_drift) to give the size of the constraint. I'm also not checking the size of the output, for performance reasons: I expect the user to be reasonable about what he is doing.

Something that could be done would be to check at attach the size of the matrices returned (jacobian should be of size nc x model.vv, drift of size nc, where nc is the constraint dimension). But there can be no guarantee that the user won't modify the size afterward (though that would be unusual I would say). I think we should choose between one of two options (currently I'm doing half of both):

 - trust the user not to change the size of the constraint on the fly. The I can do an initial size computation at attach, removing any runtime checks and allocation. Advantage: fastest solution ; inconvenient: not robust.
 - perform a full check at runtime. Advantage: allow for dynamically changing the size of the constraint by the user. Drawback: a bit slower, because I have to count the elements every time. By handling the memory well I can make it so that allocation is only performed when the size changes, but that would still bring more operations than the first solution.

Reply:
> perform a full check at runtime. Advantage: allow for dynamically changing the size of the constraint by the user. Drawback: a bit slower, because I have to count the elements every time. By handling the memory well I can make it so that allocation is only performed when the size changes, but that would still bring more operations than the first solution.

Definitely this solution for now. Checking the size should be extremely cheep (and if not we should make sure it is) compare to allocating memory.